### ISSUE TITLE
Darwin CF framework shenanigans don't help with python and interpreters

### ISSUE THREAD

**Issue by copumpkin** [NixOS/nixpkgs#31865]
The fix @LnL7 added for the CF.framework rpath generally works fine, but can still hurt us with interpreters and other dlopen-based references. I'm currently dealing with an issue around `pythonPackages.hupper`'s testsuite importing `pythonPackages.watchdog` and segfaulting as a result. For now I'm just going to take out the `watchdog` reference on Darwin, but we need a better solution in general.

**Comment by LnL7** [comment]
Yeah, that's why I didn't close https://github.com/NixOS/nixpkgs/issues/12346 yet.

**Comment by qweered** [comment]
Still relevant as of 2026-05. The original `hupper`/`watchdog` workaround is still in tree:

- `pkgs/development/python-modules/hupper/default.nix` disables `watchdog` on Darwin with a `FIXME` explicitly pointing to this issue.
- `pkgs/development/python-modules/watchdog/default.nix` carries ~20 Darwin-specific test exclusions, several annotated as segfaults.

#12346 was closed as completed in 2025-09 but the dlopen/interpreter case described here is clearly not covered by that fix.
